### PR TITLE
Added support for parsing and comparing csv logs from multiple direco…

### PR DESCRIPTION
**Pull request by nolyn**
Adding support for parsing CSV files from multiple direcotries, i.e. one CSV file per simulation (of whole testset).

### CODE REVIEW

**Comment by JensAc** [comment]
I think there is some similar functionality on the [development branch](https://github.com/IENT/RDPlot/tree/development). I think we should either merge only one of the approaches into master, or merge both approaches before merging into master. Could you try the development branch and check, whether it mimics your ideas here?

File: src/rdplot/SimulationDataItemClasses/CsvLogs.py [hunk 1]
```diff
@@ -46,13 +41,20 @@ def __init__(self, config, header, line):
...
+        # use parsed info to set unique identifier for item
+        # not actually a real path like in the other simulation item classe!
+        self.path = path + f":{self.sequence}/{self.qp}"
 
         data = {}
         for i in range(0, len(header)):
             if i in [sequence_idx, qp_idx, rate_idx]:
                 continue
-            data[header[i]] = [(rate, float(line[i]))]
+            try:
```
Comment (nolyn) [inline src/rdplot/SimulationDataItemClasses/CsvLogs.py:54]: this protects against colums in the csv that are not for rdplot. for example exit status as a string

**Comment by nolyn** [comment]
I tried with the development branch and the example csv I got from you a while ago. I was not able to open it from mutliple directories (I just copied it to some dummy directories). Notably rdplot has a completely wrong path for the file displayed in the list of plotted files. it displays it as if they file is located in the src directory of rdplot, which it is not.  i guess this clashes when i open it several times from other directories

**Comment by JensAc** [comment]
Actually, the reason for the wrong patch is that csv files are a hack. The initial design of RDPlot was built on the assumption that one file defines one simulation data item. This is totally different for csv files, as multiple simulation data items are located in the same file. Therefore, I decided to display not file paths in the plotted files widgets but other strings identifying the currently plotted simulation data items.  

I will check, how I can merge this.

**Comment by JensAc** [comment]
Ok, I checked out the PR and tried once with two csv files in separate directories. This works fine and the directory name is recognized as config. 
If I open multiple files from the same directory, these files are merged and only one RD curve is plotted. I think this is not really the way it was intended.
Is there any very good argument for merging this anyway?

**Comment by nolyn** [comment]
Hm, i didn't check that use case. Without the patch, can you then put multiple csv files for different simulations in the same directory to load them? I didn't try that since it does not conform to how the rest of rdplot operates. I think I'll just stick to using the xml file interface then. Just close the PR then if support for multiple directories is not desired.

**Comment by JensAc** [comment]
Yes, this is possible. You could check with the test files located [here](https://github.com/IENT/RDPlot/tree/master/src/rdplot/tests/test_logs/exampleCsv). 
The idea with csv files was that it defines an option to have multiple RD-curves in a single file. This seems to be a use case, if you just want to take a quick look on some RD-curves and BD measurements.

**Comment by nolyn** [comment]
Sorry for taking so long to have a look at it again. I tried it with master and development branch, drag&drop or menu open file. but i can;t open the test files. Anyway, i'm not really using this feature anymore, just using datlogs now. i'll just close this

**Pull request closed without merging**